src/planscape/forsys/views.py

```diff
@@ -3,43 +3,321 @@
+  return base_raster_copy
+
+def fetch_condition_rasters(priorities, region, project_area):
+  kScale = [300.0, -300.0]
+  kSkew = [0, 0]
+  kSrid = 9822
+
+  all_rasters = {}
+
+  for p in priorities:
+    condition_rasters = get_condition_rasters(p, region)
+    origin = [sys.float_info.max, sys.float_info.min]
+    rfinal = GDALRaster({
+      "width": 1,
+      "height": 1,
+      "srid": kSrid,
+      "origin": origin,
+      "scale": kScale,
+      "skew": kSkew,
+      "bands": [{"nodata_value": np.nan}]
+    })
+
+    for cr in condition_rasters:
+      r = cr.raster
+
+      # Checking for overlapping extents is faster than issuing a query that checks for overlaps. 
+      if not raster_extent_overlaps_project_area(r, project_area):
+        continue
+      rfinal = mosaic_rasters(rfinal, r)
+
+    all_rasters[p] = rfinal
+
+  return all_rasters
+
+def get_condition_data(raster, polygon):
+  data = {"mean": 0, "count": 0}
+  e = polygon.extent
+  o = raster.origin
+  s = raster.scale
+
+  if s[0] > 0:
+    min_x = int(np.floor((e[0] - o[0]) / s[0]))
+    max_x = int(np.ceil((e[2] - o[0]) / s[0]))
+  else:
+    min_x = int(np.floor((e[2] - o[0]) / s[0]))
+    max_x = int(np.ceil((e[0] - o[0]) / s[0]))
+
+  if s[1] > 0:
+    min_y = int(np.floor((e[1] - o[1]) / s[1]))
+    max_y = int(np.ceil((e[3] - o[1]) / s[1]))
+  else:
+    min_y = int(np.floor((e[3] - o[1]) / s[1]))
+    max_y = int(np.ceil((e[1] - o[1]) / s[1]))
+
+  data = raster.bands[0].data()
+
+  sum = 0
+  count = 0
+  for y in range(min_y, max_y + 1, 1):
+    for x in range(min_x, max_x + 1, 1):
+      d = data[y][x]
+      if np.isnan(d):
+        continue
+
+      # TODO: adjust these equations for the case where skew != [0, 0]
+      xpoly = o[0] + x*s[0]
+      ypoly = o[1] + y*s[1] 
+      p = Point((xpoly, ypoly))
+      if not p.within(polygon):
+        continue
+
+      sum = sum + d
+      count = count + 1
+
+  if count == 0:
+    return {"mean": 0, "count": 0}
+  return {"mean": sum/count, "count": count}
+
+def transform_into_forsys_df_data(condition_rasters, boundaries, project_area):
+  kConditionPrefix = "cond"
+  kPriorityPrefix = "p"
+  kUnitAreaCost = 5000
+
+  data = {}
+  data['proj_id'] = []
+  data['stand_id'] = []
+  data['shape_name'] = []
+  for c in condition_rasters.keys():
+    data[kConditionPrefix + "_" + c] = []
+    data[kPriorityPrefix + "_" + c] = []
+    data['area'] = []
+    data['cost'] = []
+
+  for b in boundaries:
+    # raster boundary seems to be 80 units off from the huc-12 boundaries.
+    # TODO: adjust raster data.
+    geo = project_area.intersection(b.geometry)
+    geo = translate_polygon(geo, 80, 0)
+    geo.transform(CoordTransform(SpatialReference(4269), SpatialReference(9822)))
+
+    # TODO: set project_area ID from an external source
+    data['proj_id'].append(1)
+    # TODO: double-check that it makes sense to use this ID.
+    data['stand_id'].append(b.id)
+    # TODO: double-check that this field is necessary.
```

Comment:
From the ForSys side we don't need it. We have `proj_id` and multi-part polygons are not a problem because the data is converted to table format at that stage already.

Reply:
Do you mean R scripts exist for this?
There's a difference between an R script and an implementation in production.